src/server/routes/editor.tsx

```diff
@@ -179,18 +180,31 @@ router.post('/edit/handler', auth.isAuthenticatedForHandler, (req, res) => {
 });
 
 router.post('/privs/edit/handler', auth.isAuthenticatedForHandler, auth.isAuthorized(ADMIN), async (req, res, next) => {
-	const {Editor} = req.app.locals.orm;
+	const {Editor, AdminLog, bookshelf} = req.app.locals.orm;
+	const {adminId, newPrivs, note, oldPrivs, targetUserId} = req.body;
+	const actionType = AdminActionType.CHANGE_PRIV;
 	try {
+		const trx = await bookshelf.transaction();
 		const editor = await Editor
-			.forge({id: parseInt(req.body.targetUserId, 10)})
-			.fetch({require: true})
+			.forge({id: parseInt(targetUserId, 10)})
+			.fetch({require: true, transacting: trx})
 			.catch(Editor.NotFoundError, () => next(new error.NotFoundError('Editor not found', req)));
-		if (editor.get('privs') === req.body.newPrivs) {
+		if (editor.get('privs') === newPrivs) {
 			return next(new error.FormSubmissionError(
 				'No change to Privileges', req
 			));
 		}
-		await editor.save({privs: req.body.newPrivs});
+		await editor.save({privs: newPrivs});
```

Comment:
It's for this save operation that we really need the SQL transaction:
```suggestion
		await editor.save({privs: newPrivs},{transacting: trx});
```